src/compas_timber/planning/sequencer.py

```diff
@@ -0,0 +1,274 @@
+        super(Model3d, self).__init__(id, location)
+        self.geometry = geometry
+        self.element_id = element_id
+        self.obj_filepath = obj_filepath
+
+    @property
+    def data(self):
+        data_dict = {
+            "geometry": self.geometry,
+            "element_id": self.element_id,
+            "obj_filepath": self.obj_filepath,
+        }
+        data_dict.update(super(Model3d, self).data)
+        return data_dict
+
+    def transform(self, tranformation):
+        super(Model3d, self).transform(tranformation)
+        self.geometry.transform(tranformation)
+
+
+class Text3d(Instruction):
+    """Text overlay"""
+
+    def __init__(self, id, location, text, size):
+        super(Text3d, self).__init__(id, location)
+        self.text = text
+        self.size = size
+
+    @property
+    def data(self):
+        data_dict = {
+            "text": self.text,
+            "size": self.size,
+        }
+        data_dict.update(super(Text3d, self).data)
+        return data_dict
+
+
+class LinearDimension(Instruction):
+    """3d linear dimension"""
+
+    def __init__(self, id, location, start, end, char_size, offset):
+        super(LinearDimension, self).__init__(id, location)
+        self.start = start
+        self.end = end
+        self.char_size = char_size
+        self.offset = offset
+
+    @property
+    def data(self):
+        data_dict = {
+            "start": self.start,
+            "end": self.end,
+            "char_size": self.char_size,
+            "offset": self.offset,
+        }
+        data_dict.update(super(LinearDimension, self).data)
+        return data_dict
+
+    def transform(self, tranformation):
+        super(LinearDimension, self).transform(tranformation)
+        self.start.transform(tranformation)
+        self.end.transform(tranformation)
+
+
+class Step(Data):
+    """Container for building instructions which assemble a single element
+
+    Attributes
+    ----------
+    location : :class:`compas.geometry.Frame`  # TODO: replace with Transformation?
+        Location of the step.
+    obj_filepath : str
+        Path to the 3d model of the element(s) which belong to this step.
+    instructions : list(:class:`Instruction`)
+        List of instructions which make up the step.
+    element_ids : list(int)
+        List of cad element ids which are associated with the step.
+
+    """
+
+    def __init__(
+        self,
+        element_ids,
+        location=None,
+        obj_filepath=None,
+        step_id=None,
+        instructions=None,
+        is_built=False,
+        actor=None,
+    ):
+        super(Step, self).__init__()
+        self.element_ids = element_ids or []
+        self.location = location or Frame.worldXY()
+        self.obj_filepath = obj_filepath
+        self.step_id = step_id
+        self.instructions = instructions or []
+        self.is_built = is_built
+        self._actor = actor
+
```

Comment:
add `objects_held` - list of beam keys which are being held by the robot during the execution of this step.
add `priority` - steps with the same priority value can be interchangeable.